vscode-smart-contracts/src/extension.ts

```diff
@@ -1,21 +1,141 @@
 /*
  * This module is the entrypoint of the extension.
  * The exported functions have special meaning and are called by VS Code.
- *
  */
 
 import * as vscode from "vscode";
 import * as Commands from "./commands";
+import * as util from "node:util";
+import * as globCallback from "glob";
+import * as cargoConcordium from "./cargo-concordium";
+import * as path from "node:path";
+import {
+  CONCORDIUM_TASK_TYPE,
+  ConcordiumTaskDefinition,
+} from "./cargo-concordium";
+
+const glob = util.promisify(globCallback);
 
 // This method is called when the extension is activated
 // An extension is activated the very first time a command is executed.
 export function activate(context: vscode.ExtensionContext) {
   // The command has been defined in the package.json file
   // Now provide the implementation of the command with registerCommand
   // The commandId parameter must match the command field in package.json
-  const disposable = vscode.commands.registerCommand(
-    "concordium-smart-contracts.version",
-    Commands.version
+  context.subscriptions.push(
+    vscode.commands.registerCommand(
+      "concordium-smart-contracts.version",
+      Commands.version
+    ),
+    vscode.commands.registerTextEditorCommand(
+      "concordium-smart-contracts.build",
+      Commands.build
+    ),
+    vscode.commands.registerTextEditorCommand(
+      "concordium-smart-contracts.build-skip-schema",
+      Commands.buildSkipSchema
+    ),
+    vscode.commands.registerTextEditorCommand(
+      "concordium-smart-contracts.build-embed-schema",
+      Commands.buildEmbedSchema
+    ),
+    vscode.commands.registerTextEditorCommand(
+      "concordium-smart-contracts.test",
+      Commands.test
+    ),
+    vscode.tasks.registerTaskProvider(CONCORDIUM_TASK_TYPE, taskProvider)
   );
-  context.subscriptions.push(disposable);
+}
+
+/**
+ * Task provider for Concordium tasks.
+ */
+const taskProvider: vscode.TaskProvider = {
+  /** Search the current workspace for possible tasks */
+  async provideTasks() {
+    const workspaceFolders = vscode.workspace.workspaceFolders ?? [];
+    const tasks = await Promise.all(
+      workspaceFolders.map(async (workspaceFolder) => {
+        const workspaceRoot = workspaceFolder.uri.fsPath;
+        const cargoProjectDirs = await getCargoProjectDirs(workspaceRoot);
+        return Promise.all(
+          cargoProjectDirs.flatMap((cwd) => {
+            const defaultOutDir = path.join(cwd, "out");
```

Comment:
Have you considered using `concordium-out` as the name instead? To make it less likely to clash?

Reply:
clash with what? this will be in the contract directory, so it seems fine to assume mainly cargo stuff is here

Reply:
I often create "out" directories for stuff. It's such a generic name.

Reply:
Okay, I'll use `concordium-out` then